brpc-java-core/src/main/java/com/baidu/brpc/protocol/dubbo/ClassUtils.java

```diff
@@ -0,0 +1,284 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.baidu.brpc.protocol.dubbo;
+
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+public class ClassUtils {
+    /**
+     * Suffix for array class names: "[]"
+     */
+    public static final String ARRAY_SUFFIX = "[]";
+    /**
+     * Prefix for internal array class names: "[L"
+     */
+    private static final String INTERNAL_ARRAY_PREFIX = "[L";
+    /**
+     * Map with primitive type name as key and corresponding primitive type as
+     * value, for example: "int" -> "int.class".
+     */
+    private static final Map<String, Class<?>> PRIMITIVE_TYPE_NAME_MAP = new HashMap<String, Class<?>>(16);
+    /**
+     * Map with primitive wrapper type as key and corresponding primitive type
+     * as value, for example: Integer.class -> int.class.
+     */
+    private static final Map<Class<?>, Class<?>> PRIMITIVE_WRAPPER_TYPE_MAP = new HashMap<Class<?>, Class<?>>(8);
+
+    private static final char PACKAGE_SEPARATOR_CHAR = '.';
+
+    static {
+        PRIMITIVE_WRAPPER_TYPE_MAP.put(Boolean.class, boolean.class);
+        PRIMITIVE_WRAPPER_TYPE_MAP.put(Byte.class, byte.class);
+        PRIMITIVE_WRAPPER_TYPE_MAP.put(Character.class, char.class);
+        PRIMITIVE_WRAPPER_TYPE_MAP.put(Double.class, double.class);
+        PRIMITIVE_WRAPPER_TYPE_MAP.put(Float.class, float.class);
+        PRIMITIVE_WRAPPER_TYPE_MAP.put(Integer.class, int.class);
+        PRIMITIVE_WRAPPER_TYPE_MAP.put(Long.class, long.class);
+        PRIMITIVE_WRAPPER_TYPE_MAP.put(Short.class, short.class);
+
+        Set<Class<?>> primitiveTypeNames = new HashSet<Class<?>>(16);
+        primitiveTypeNames.addAll(PRIMITIVE_WRAPPER_TYPE_MAP.values());
+        primitiveTypeNames.addAll(Arrays
+                .asList(boolean[].class, byte[].class, char[].class, double[].class,
+                        float[].class, int[].class, long[].class, short[].class));
+        for (Class<?> primitiveTypeName : primitiveTypeNames) {
+            PRIMITIVE_TYPE_NAME_MAP.put(primitiveTypeName.getName(), primitiveTypeName);
+        }
+    }
+
+    public static Class<?> forNameWithThreadContextClassLoader(String name)
+            throws ClassNotFoundException {
+        return forName(name, Thread.currentThread().getContextClassLoader());
+    }
+
+    public static Class<?> forNameWithCallerClassLoader(String name, Class<?> caller)
+            throws ClassNotFoundException {
+        return forName(name, caller.getClassLoader());
+    }
+
+    public static ClassLoader getCallerClassLoader(Class<?> caller) {
+        return caller.getClassLoader();
+    }
+
+    /**
+     * get class loader
```

Comment:
We should add more detail comments for this method because the behaviors of this method is too complex. Otherwise users of this method never know the behaviors if they don't read the code.